# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/util/ActiveElementManager.h
# Commit: 024cf173d87d
# Full Hash: 024cf173d87dbc5d3b581d2b121e3d8295e4f46f
# Author: Dan Robertson <drobertson@mozilla.com>
# Date: 2023-09-26 09:27:49
# Regressor Bug: 1816473
# File Overlap Count: 3
# Description:
#   Bug 1816473 - Do not delay dispatching synthesized mouse events. r=botond,edgar
#   
#   Do not delay dispatching synthesized mouse events for a single tap
#   gesture.
#   
# ==============================================================================

diff -r 37eee8ccc556 -r 024cf173d87d gfx/layers/apz/util/ActiveElementManager.h
--- a/gfx/layers/apz/util/ActiveElementManager.h	Mon Sep 25 19:08:05 2023 +0000
+++ b/gfx/layers/apz/util/ActiveElementManager.h	Mon Sep 25 19:13:07 2023 +0000
@@ -21,6 +21,8 @@
 
 namespace layers {
 
+class DelayedClearElementActivation;
+
 /**
  * Manages setting and clearing the ':active' CSS pseudostate in the presence
  * of touch input.
@@ -61,6 +63,14 @@
    * delayed until after touch listeners have responded to the APZ.
    */
   void HandleTouchEnd();
+  /**
+   * Possibly clear active element sate in response to a single tap.
+   */
+  void ProcessSingleTap();
+  /**
+   * Cleanup on window destroy.
+   */
+  void Destroy();
 
  private:
   /**
@@ -82,6 +92,10 @@
    */
   RefPtr<CancelableRunnable> mSetActiveTask;
 
+  // Store the pending single tap event element activation clearing
+  // task.
+  RefPtr<DelayedClearElementActivation> mDelayedClearElementActivation;
+
   // Helpers
   void TriggerElementActivation();
   void SetActive(dom::Element* aTarget);